pilot/pkg/config/coredatamodel/controller.go

```diff
@@ -176,18 +179,26 @@ func (c *Controller) Apply(change *sink.Change) error {
 	}
 
 	var prevStore map[string]map[string]*model.Config
-
 	c.configStoreMu.Lock()
 	prevStore = c.configStore[descriptor.Type]
 	c.configStore[descriptor.Type] = innerStore
 	c.configStoreMu.Unlock()
 
-	if descriptor.Type == model.ServiceEntry.Type {
+	if c.options.IncrementalEDS && descriptor.Collection == model.SyntheticServiceEntry.Collection {
+		if err := c.edsUpdate(innerStore, prevStore); err != nil {
+			return err
+		}
+	}
+
+	if !c.options.IncrementalEDS && descriptor.Type == model.ServiceEntry.Type { // TODO(Nino-k): remove this case once incrementalUpdate is default
 		c.serviceEntryEvents(innerStore, prevStore)
-	} else {
-		c.options.ClearDiscoveryServerCache()
+		return nil
 	}
 
+	if versionChange(innerStore, prevStore) {
```

Comment:
I'm not sure we need this?

Reply:
I wasn't 100% sure either, but it can essentially safeguard against a client who sends same version repeatedly.